### PR TITLE
Add code first NestJS tests

### PR DESCRIPTION
**Pull request by hwillson**
These tests validate that NestJS has 100% Federation 2 coverage using their code first approach.

I'll keep this in draft until we figure out how we want to represent projects that have tests for both code and schema first, but the good news is that as of [`@nestjs/graphql@10.1.7`](https://github.com/nestjs/graphql/releases/tag/v10.1.7) NestJS fully supports Fed 2 with both approaches.

Fixes https://github.com/apollographql/apollo-federation-subgraph-compatibility/issues/296

### CODE REVIEW

**Comment by hwillson** [comment]
For now I've just duplicated the NestJS section and added schema first / code first beside the project name:

CleanShot 2022-12-10 at 16 15 39

Maybe we can run with this for now since NestJS is the only project we're currently showing schema and code tests for, but at some point (soon) we should revisit this breakdown.

**Comment by hwillson** [comment]
On that note, this is now ready for review.

File: implementations/nestjs-code-first/.gitignore [hunk 1]
```diff
@@ -0,0 +1,398 @@
+# Created by .ignore support plugin (hsz.mobi)
```
Comment (dariuszkuc) [inline implementations/nestjs-code-first/.gitignore:1]: this is a pretty large `.gitignore` file -> should we trim it down to be specific to this impl? or update root ignore file with additional configs (if needed)?

File: implementations/nestjs-code-first/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+services:
+  products:
+    # must be relative to the root of the project
+    build: implementations/nestjs
```
Comment (dariuszkuc) [inline implementations/nestjs-code-first/docker-compose.yaml:4]: should this be `implementations/nestjs-code-first`?

File: implementations/nestjs-code-first/package.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    "start:prod": "node dist/main"
+  },
+  "dependencies": {
+    "@apollo/subgraph": "^2.2.2",
+    "@nestjs/apollo": "^10.1.7",
+    "@nestjs/common": "^9.2.1",
+    "@nestjs/core": "^9.2.1",
+    "@nestjs/graphql": "^10.1.7",
+    "@nestjs/platform-express": "^9.2.1",
+    "apollo-server-express": "^3.10.0",
```
Comment (dariuszkuc) [inline implementations/nestjs-code-first/package.json:23]: should we use AS4 instead?

Comment (dariuszkuc) [reply implementations/nestjs-code-first/package.json:23]: i.e. https://www.npmjs.com/package/@as-integrations/next

File: implementations/nestjs-code-first/.gitignore (hunk 1 above)
Comment (hwillson) [reply implementations/nestjs-code-first/.gitignore:1]: For sure - this is the `.gitignore` Nest's bootstrap project includes. Happy to trim it down!

File: implementations/nestjs-code-first/package.json (hunk 3 above)
Comment (hwillson) [reply implementations/nestjs-code-first/package.json:23]: NestJS hasn't been updated to use AS4 yet unfortunately. That work is being tracked here: https://github.com/nestjs/graphql/issues/2445

File: implementations/nestjs-code-first/docker-compose.yaml (hunk 2 above)
Comment (hwillson) [reply implementations/nestjs-code-first/docker-compose.yaml]: It should - fixed! I've been verifying this work outside of Docker so everything will still pass after adjusting this.

File: implementations/nestjs-code-first/.gitignore (hunk 1 above)
Comment (hwillson) [reply implementations/nestjs-code-first/.gitignore]: I just killed it completely - we don't need any of it. I'm not sure why all of that is in their bootstrap project (are folks really using IntelliJ with NestJS? 🤷).

**Review by dariuszkuc — APPROVED** [review]

**Pull request merged**